### PR TITLE
added tests for network components node and edge

### PR DESCRIPTION
**Pull request by MichaelMarien**
Was looking into a strange issue with adding PyTorch backend. Decided to debug and add more tests. As we were missing very basic unit tests add the moment, made separate PR for a first batch of simple tests for network components Node and Edge. Two questions I have that I encountered writing tests:

* when giving a tensor to a Node for initialisation, should we not convert it using the backend? In typical use cases, Node would not be called directly and the network will do the conversion to the backend type, but still.

* get_dimension method of Node class specifies in docstring it raises and error if input is not an int. I believe it can be a str too? Can quickly change docstring in this PR if so.

### CODE REVIEW

File: tensornetwork/network_components_test.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+
+def test_edge_initialize_dangling(single_node_edge):
+    node = single_node_edge.node
+    edge = single_node_edge.edge
+    assert edge.name == "edge"
+    assert edge.node1 == node
+    assert edge.axis1 == 0
+    assert edge.node2 is None
+    assert edge.axis2 is None
+    assert edge._is_dangling is True
```
[the remark this answers is not in the document]
Comment (Thenerdstation) [reply tensornetwork/network_components_test.py]: Similar note to `node._tensor`. Prefer to use the `node.tensor` property

File: tensornetwork/network_components_test.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+def test_edge_is_dangling(double_node_edge):
+    edge1 = double_node_edge.edge1
+    edge12 = double_node_edge.edge12
+    assert edge1.is_dangling()
+    assert not edge12.is_dangling()
+
+
+def test_edge_is_trace_true(single_node_edge):
+    node = single_node_edge.node
+    edge = Edge(name="edge", node1=node, axis1=2, node2=node, axis2=2)
```
Comment (Thenerdstation) [inline tensornetwork/network_components_test.py]: We should never  have an edge point the the same axis twice one the same node. Can you change the axis value here? (We should also raise an error if this happens, but we can do that in a later PR)
